io_scene_godot/converters/mesh.py

```diff
@@ -151,17 +159,126 @@ def make_arrays(escn_file, export_settings, node, armature_data):
         gid_to_bid_map
     )
 
+    if (export_settings['export_shape_key'] and
+            node.data.shape_keys is not None):
+        export_morphs(
+            export_settings, node, surfaces, has_tangents, gid_to_bid_map)
+
     has_bone = True if armature_data is not None else False
 
     for surface_id, surface in enumerate(surfaces):
         surface.id = surface_id
-        surface.has_bone = has_bone
+        surface.vertex_data.has_bone = has_bone
+        for vert_data in surface.morph_arrays:
+            vert_data.has_bone = has_bone
 
     bpy.data.meshes.remove(mesh)
 
+    if armature_data is not None:
+        armature_data.pose_position = original_pose_position
+        bpy.context.scene.update()
+
     return surfaces
 
 
+def extract_shape_keys(blender_shape_keys):
+    """Return a list of (shape_key_index, shape_key_object) each of them
+    is a shape key needs exported"""
+    # base shape key needn't be exported
+    ret = list()
+    base_key = blender_shape_keys.reference_key
+    for index, shape_key in enumerate(blender_shape_keys.key_blocks):
+        if shape_key != base_key:
+            ret.append((index, shape_key))
+    return ret
+
+
+def export_morphs(export_settings, node, surfaces, has_tangents,
+                  gid_to_bid_map):
+    """Export shape keys in mesh node and append them to surfaces"""
+    for index, shape_key in extract_shape_keys(node.data.shape_keys):
+
+        node.show_only_shape_key = True
+        node.active_shape_key_index = index
+        shape_key.value = 1.0
+
+        mesh = node.to_mesh(bpy.context.scene,
+                            export_settings['use_mesh_modifiers'],
+                            "RENDER")
+        triangulate_mesh(mesh)
+
+        if has_tangents:
+            mesh.calc_tangents()
+        else:
+            mesh.calc_normals_split()
+
+        surface_to_morph = [None] * len(surfaces)
```

Comment:
It looks to me like there are multiple mappings going on here, one from material to surface, and one from surface to a `VerticesArrays()`. I think these mappings could be generated:
 - before iterating through the faces
 - In a separate function.

They could possibly take the form of:
{material_index: [surface, VerticesArray]}

Reply:
true, when I look at it now, I got a little dizzy..